src/components/FeaturesDemo/index.tsx

```diff
@@ -0,0 +1,726 @@
+import {
+    pause,
+    ScreenOrientation,
+    scrollToLineIfNeeded,
+    typeCode,
+    typeConsoleMessage,
+    useScreenOrientation,
+} from "@site/src/components/FeaturesDemo/utils";
+
+export function FeaturesDemo(props: { className?: string }) {
+    const store = useStore();
+    const heading = useStore(state => state.heading);
+
+    const appBrowserPageRef = useRef<WebBrowserApi>(null);
+    const reportBrowserPageRef = useRef<WebBrowserApi>(null);
+    const reportCursorRef = useRef<AnimatedCursorApi>(null);
+
+    const codeRefs = CODE_MESSAGES.map(section => ({
+        ...section,
+        ref: useRef<HTMLSpanElement>(null),
+    }));
+    const codeElementRef = useRef<HTMLDivElement>(null);
+    const codeCleanupRef = useRef<() => void>(null);
+
+    const codeElement = (
+        <div
+            className="relative h-full w-full overflow-y-scroll scroll-smooth p-1"
+            ref={codeElementRef}
+        >
+            <div
+                className="absolute text-right text-blue-400 opacity-40"
+                dangerouslySetInnerHTML={{
+                    __html: Array.from(Array(50).keys()).join("<br>"),
+                }}
+            ></div>
+            <div className="pl-7">
+                {codeRefs.map(({ ref }, index) => (
+                    <span ref={ref} key={index}></span>
+                ))}
+            </div>
+        </div>
+    );
+
+    const consoleRefs = CONSOLE_MESSAGES.map(message => ({
+        ...message,
+        ref: useRef<HTMLSpanElement>(null),
+    }));
+    const consoleCleanupRef: MutableRefObject<(() => void) | null> = useRef<() => void>(null);
+
+    const consoleElement = (
+        <>
+            {consoleRefs.map(({ messages, ref }, index) => (
+                <span ref={ref} key={`${messages[0]}_${index}`}></span>
+            ))}
+        </>
+    );
+
+    const controller = new AbortController();
+    const signal = controller.signal;
+
+    const steps: (() => void | Promise<void>)[] = [
+        async function openCodeEditor() {
+            store.setHeading(getById(HEADINGS, HeadingId.FamiliarApi).data, signal);
+
+            if (useStore.getState().windows.length > 0) {
+                store.popWindow(signal);
+            }
+            store.pushWindow(
+                {
+                    key: "code-editor",
+                    type: ComponentTypes.CodeEditor,
+                    props: {
+                        tabs: SAMPLE_TABS,
+                        fsTreeRoots: SAMPLE_FS_TREE_ROOTS,
+                        code: codeElement,
+                        console: consoleElement,
+                    },
+                },
+                signal,
+            );
+
+            await pause(PAUSE_SM, signal);
+        },
+        async function typeBasicTest() {
+            await typeCode(getById(codeRefs, CodeId.BasicTestBegin), codeCleanupRef);
+            await typeCode(getById(codeRefs, CodeId.BasicTestEnd), codeCleanupRef);
```

Comment:
That's because of how `Typed.js` works. As you can see, in our use case, we need the following features:
- type some code
- go back to the middle of the code
- remove some code and type another one (for fixing input selector, for instance)

However, `Typed.js` has only one feature: type some code into an element. It doesn't let you type in the middle or move the cursor in any way. So, in order to circumvent this constraint, we split the code into spans like this: `BasicTestBegin|ReplTestBegin|...|ReplTestEnd|BasicTestEnd`, then we can fill those spans in any order we want, giving the illusion of typing in the middle.

I've spent a while trying to come up with a better solution for this case, but it seems like this is our best bet. There's more versatile option out there — https://www.typeitjs.com/, but it requires a paid license. Implementing this typing experience from scratch is not trivial either, I've tried it too.